### ISSUE TITLE
The conversion of the selector loc will cause errors in multi-line cases after `eslint@>=9.30.0`

### ISSUE THREAD

**Issue by CyanSalt** [future-architect/eslint-plugin-vue-scoped-css#397]
Since I can't reproduce this issue on the Playground, I created a simple example at https://github.com/CyanSalt/vue-scoped-css-repro

Note that `postcss-selector-parser` actually produces column numbers of 0, for example:

```js
const parser = require('postcss-selector-parser')
parser().astSync('.foo,\n.bar').nodes.map(node => node.source)
// [
//   { start: { line: 1, column: 1 }, end: { line: 1, column: 4 } },
//   { start: { line: 2, column: 0 }, end: { line: 2, column: 4 } }
// ]
```

This will cause `getESLintLineAndColumnFromPostCSSSelectorParserNode` to convert it into a negative column.

https://github.com/future-architect/eslint-plugin-vue-scoped-css/blob/327aae778e07279e595c3503555e746e4f152d09/lib/styles/parser/selector/css-selector-parser.ts#L613-L635

which will be rejected by `SourceCode#getIndexFromLoc`.

https://github.com/eslint/eslint/blob/main/lib/languages/js/source-code/source-code.js#L820-L824

**Comment by ota-meshi** [comment]
Thanks for posting the report!
It seems the issue only occurs with ESLint v9.30 and later.
